play/src/front/Streaming/BroadcastService.ts

```diff
@@ -23,42 +35,132 @@ export class BroadcastService {
+        this.broadcastSpaces = [];
 
         this.megaphoneEnabledUnsubscribe = megaphoneEnabledStore.subscribe((megaphoneEnabled) => {
             if (megaphoneEnabled) {
-                this.jitsiConference?.broadcast(["video", "audio"]);
+                this.broadcastSpaces.forEach((broadcastSpace) => {
+                    broadcastSpace.jitsiConference?.broadcast(["video", "audio"]);
+                });
             } else {
-                this.jitsiConference?.broadcast([]);
+                this.broadcastSpaces.forEach((broadcastSpace) => {
+                    broadcastSpace.jitsiConference?.broadcast([]);
+                });
             }
         });
 
-        libJitsiFactory
-            .createConnection(
-                "jitsi.test.workadventu.re",
-                "prosody.test.workadventu.re",
-                "muc.prosody.test.workadventu.re"
-            )
-            //.createConnection("coremeet.workadventu.re", "prosody.workadventu.re", "muc.prosody.workadventu.re")
-            .then(async (connection) => {
-                this.jitsiConference = await JitsiConferenceWrapper.join(connection, "conferencetestfromjitsilib");
-                jitsiConferencesStore.set("conferencetestfromjitsilib", this.jitsiConference);
+        //libJitsiFactory.createConnection("jitsi.test.workadventu.re", "prosody.test.workadventu.re", "muc.prosody.test.workadventu.re").then((connection) => {
+    }
 
-                if (get(megaphoneEnabledStore)) {
-                    this.jitsiConference.broadcast(["video", "audio"]);
+    public joinSpace(spaceName: string) {
+        const spaceFilter = this.connection.emitWatchSpaceLiveStreaming(spaceName);
+        const broadcastSpace = new BroadcastSpace(this.connection, spaceName, spaceFilter);
+        broadcastSpace.unsubscribes.push(
+            broadcastSpace.users.subscribe((users) => {
+                if (users.size === 0) {
+                    if (broadcastSpace.jitsiConference !== undefined) {
+                        const jitsiConference = broadcastSpace.jitsiConference;
+                        this.limit(() => jitsiConference.leave())
+                            .then(async () => {
+                                broadcastSpace.jitsiConference = undefined;
+                                if (this.canDisconnect()) {
+                                    console.log("Disconnecting from Jitsi");
+                                    await this.jitsiConnection?.disconnect();
+                                    this.jitsiConnection = undefined;
+                                }
+                            })
+                            .catch((e) => {
+                                console.error(e);
+                            });
+                    }
+                } else {
+                    if (broadcastSpace.jitsiConference === undefined) {
+                        this.limit(() => this.joinJitsiConference(spaceName, broadcastSpace))
```

Comment:
The `if (broadcastSpace.jitsiConference === undefined)` should be inside the callback (otherwise the conf could have been started when we wait for the "limit"